src/dmd/expressionsem.d

```diff
@@ -6754,19 +6754,20 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                 {
                     // lower to `object.__ArrayCast!(TFrom, TTo)(from)`
 
-                    auto id = Id.__ArrayCast;
+                    // fully qualify as `object.__ArrayCast`
+                    Expression id = new IdentifierExp(exp.loc, Id.empty);
+                    auto dotid = new DotIdExp(exp.loc, id, Id.object);
 
                     auto tiargs = new Objects();
                     tiargs.push(tFrom);
                     tiargs.push(tTo);
-                    auto ti = new TemplateInstance(exp.loc, id, tiargs);
-                    Expression __ArrayCast = new ScopeExp(exp.loc, ti);
+                    auto dt = new DotTemplateInstanceExp(exp.loc, dotid, Id.__ArrayCast, tiargs);
 
                     auto arguments = new Expressions();
                     arguments.push(exp.e1);
-                    __ArrayCast = new CallExp(exp.loc, __ArrayCast, arguments);
+                    Expression ce = new CallExp(exp.loc, dt, arguments);
```

Comment:
`result = new CallExp(exp.loc, dt, arguments). expressionSemantic(sc)` would save a variable.

Reply:
I don't care for that for the following reason.  With DMD I always have to put `printf` statements in strategic places to learn what DMD is doing and try to understand what's going on.  If the two statements are separate, I can easily throw in a `printf` statement for a before-and-after comparison to inspect what DMD is doing in the semantic phase.  I've had to break up compound expressions before to do this, and it's a bit of nuisance.

Reply:
Ok.